mopidy_musicbox_webclient/static/js/images.js

```diff
@@ -61,47 +64,92 @@
+                    for (var i = 0; i < data.album.image.length; i++) {
+                        if (data.album.image[i].size === size) {
+                            $(images).attr('src', data.album.image[i]['#text'] || this.DEFAULT_ALBUM_URL)
+                            break
+                        }
+                    }
+                }, error: function (code, message) {
+                    console.error('Error retrieving album info from last.fm', code, message)
+                }})
+            } else {
+                console.warn('Not enough tag information available to retrieve album cover from last.fm, using default')
+            }
+        },
+
+        getArtistInfo: function (track) {
+            var artistName = ''
+            var musicBrainzID = ''
+
+            if (track && track.album && track.album.artists && track.album.artists.length > 0) {
                 // First look for the artist in the album
-                artistname = track.album.artists[0].name
-            } else if (track.artists && (track.artists.length > 0)) {
+                artistName = track.album.artists[0].name
+                musicBrainzID = track.album.artists[0].musicbrainz_id
+            } else if (track && track.artists && (track.artists.length > 0)) {
                 // Fallback to using artists for specific track
-                artistname = track.artists[0].name
+                artistName = track.artists[0].name
+                musicBrainzID = track.artists[0].musicbrainz_id
             }
 
-            this.lastfm.album.getInfo({artist: artistname, album: albumname}, {success: function (data) {
-                for (var i = 0; i < data.album.image.length; i++) {
-                    if (data.album.image[i].size === size) {
-                        $(images).attr('src', data.album.image[i]['#text'] || defUrl)
-                    }
-                }
-            }, error: function (code, message) {
-                console.error('Error retrieving album info from last.fm', code, message)
-            }})
+            return {id: musicBrainzID, name: artistName}
         },
 
-        getArtistImage: function (artist, images, size) {
-            var defUrl = 'images/user_24x32.png'
-            $(images).attr('src', defUrl)
-            if (!artist || artist.length === 0) {
+        getArtistImage: function (uri, images, size, mopidy) {
+            $(images).attr('src', this.DEFAULT_ARTIST_URL)
+            if (!uri || uri.length === 0) {
                 return
             }
-            this.lastfm.artist.getInfo({artist: artist}, {success: function (data) {
-                for (var i = 0; i < data.artist.image.length; i++) {
-                    if (data.artist.image[i].size === size) {
-                        $(images).attr('src', data.artist.image[i]['#text'] || defUrl)
-                    }
+
+            var _this = this
+            mopidy.library.lookup({'uris': [uri]}).then(function (resultDict) {
```

Comment:
Mopidy-spotify should, it can get images for any resource I think. Mopidy-tunein handles either artists or albums (can't remember which, it's a bit arbitrary) as station and show logos. Others, not so sure.